src/object_sync/hello.cpp

```diff
@@ -39,7 +39,21 @@ namespace object_sync {
             if (info.Length() >= 1) {
               if (info[0]->IsString()) 
               {
-                std::string name = *v8::String::Utf8Value(info[0]->ToString());
+                // Don't want to risk passing a null string around, which might create unpredictable behavior.
+                Nan::Utf8String utf8_value(info[0]);
+                int len = utf8_value.length();
+                if (len <= 0) {
+                   return Nan::ThrowTypeError("arg must be a non-empty string");
+                }
+
+                // This line converts a V8 string to a C++ std::string.
+                // In the background, it triggers memory allocation (stack allocating, but std:string is also dynamically allocating memory in the heap)
+                // We want to avoid heap allocation to ensure more performant code.
+                // See https://github.com/mapbox/cpp/blob/master/glossary.md#stack-allocation
+                // and https://stackoverflow.com/questions/79923/what-and-where-are-the-stack-and-heap/80113#80113
+                // Also, providing the length allows the std::string constructor to avoid calculating the length internally 
+                // and should be faster since it skips an operation. 
+                std::string name(*utf8_value, len);
                 auto *const self = new HelloObject(name);
```

Comment:
Yes, always relevant when performance matters, so in our case, let's apply wherever possible based on the idea that developers will copy/paste from skel and this code should be fast by default so that best practices trickle into code that eventually will be processing a lot of data, where perf will matter in a realworld way.